### DISCUSSION TITLE
How can I get a success rate in yolo 5?

### DISCUSSION THREAD

**Discussion by soleymany** [ultralytics/yolov5#6224] [Q&A]
Hi everyone. I labeled a number of images and did my own tests after training with yolo5. Absolutely successful. Now I do not know how to get the position rate (Success rate)?
Please help me how can I get a success rate of rejection ???

**Comment by glenn-jocher** [comment]
'Success rate' is not an object detection metric I'm aware of. 

val.py produces P, R, F1 and mAP metrics.

**Reply by soleymany** [reply]
Thanks for your answer, in fact I  got P, R, F1 and mAP metrics, I got these metrics after my train, but I want to get these metrics after my test, is it possible????? I want to get a 'Success rate' for my project after my test. I want to make sure after my test, finally with what 'Success rate' did I do this test?? with have 'Success rate' in a test, I can understand result for my work that my work is good or no...By 'Success rate' in yolov5, I mean the following concept:
Success rate= The number of objects we are considering that have been correctly identified within the threshold we are considering/The total number of objects we consider in the total test data

**Reply by glenn-jocher** [reply]
@soleymany you can evaluate the test part of your dataset with :
```
python val.py --data dataset.yaml --task test
```

**Reply by soleymany** [reply]
Hi, I wrote your cod but I faced with this error. I send a image 
![GitHub](https://user-images.githubusercontent.com/85700057/148767962-a6742013-fc86-4795-b669-fd1ae5f4dcc5.PNG)
from it error for you.
I did my test and after test, I ran your cod.

**Reply by felix-knopp** [reply]
you made a typo in "dcustom_data.yaml".
omit the "d" and try again

**Reply by soleymany** [reply]
Oh, yes. thanks to your attention. 
I changed it, but I faced to that error again.

**Reply by felix-knopp** [reply]
move the custom_data.yaml into the yolov5/data directory or use: 
`python val.py --data ../custom_data.yaml --task test`

**Reply by soleymany** [reply]
![github2](https://user-images.githubusercontent.com/85700057/148821159-1d9d19e5-5d54-4e55-ae66-64065adc5e31.PNG)
AssertionError: Multiple files match '../custom_data.yaml', specify exact path: ['/content/yolov5/data/images/../custom_data.yaml', '/content/yolov5/data/hyps/../custom_data.yaml', '/content/yolov5/data/scripts/../custom_data.yaml']

**Reply by felix-knopp** [reply]
you moved the file AND you changed the command. 
I said 'move the file OR use the other command.'

Now that you moved it... just use `python val.py --data custom_data.yaml --task test`

**Reply by soleymany** [reply]
I changed, but I faced to this error :
![github3](https://user-images.githubusercontent.com/85700057/148838427-07f80105-4e12-4229-94f3-be49b24e942c.PNG)
I tried another command, but this error was again:
![github4](https://user-images.githubusercontent.com/85700057/148838718-810ff57b-1428-4b7c-a233-fdd26cfa419c.PNG)

**Reply by felix-knopp** [reply]
you want to validate the model that you trained. 
so for weights you most likely want the file in 'runs/expXX/weights/best.pt', where XX stands for the training run you want to validate.
try:
`python val.py --weights runs/expXX/weights/best.pt --data ../custom_data.yaml --task test` 

Dont forget to exchange XX with a number so it matches with one of the directories in runs

**Reply by soleymany** [reply]
With this command : 'python val.py --weights runs/train/exp2/weights/best.pt --data ../custom_data.yaml --task test'
I took this answer:
![github5](https://user-images.githubusercontent.com/85700057/148894781-cb0fc3ad-07d0-4e28-a882-87e3efd44498.PNG)

Do you think this metrics for map, p, and r are for after test?
 because my main goal was, i get those after my test.

**Reply by soleymany** [reply]
Thank you very much for your answer and attention, I got the answer, thank you very much.
